packages/contract/src/contracts/functions/base-invocation-scope.ts

```diff
@@ -184,6 +195,29 @@ export class BaseInvocationScope<TReturn = any> {
     return this;
   }
 
+  async addMissingVariableOutputs(
```

Comment:
We can do this.

For clarification:
`dryRun` on the `base-invocation-scope` uses `provider.call`,  I can add it to both `provider.sendTransaction` and `provider.call` functions.

Reply:
yes, the issue I just thought about on the sendTransaction is that doing on the provider, will be possible to know dryRun before and see what is missing but will not be possible to change the tx as it requires signature.

Reply:
this is correct,  `signTransaction` and `populateTransactionWitnessesSignature` are on the wallet

Reply:
@luizstacio  - I refactored `provider.call` to add the variables.  But I'm having issues with `provider.sendTransaction`.  The gas estimation + adding output variables results in test failures.  My suggestion is to merge the new `provider.call` code in this PR and open another issue to add feature to `provider.sendTransaction` so this PR doesn't get too old and we can ship smaller chunks.

Reply:
I think the idea on `sendTransaction` is to run the check for variable outputs on dry run mode with UTXO Validation false. If it doesn't have variable outputs just throw an error. If tests fail would be because of missing variable outputs. Otherwise, it should keep passing no?